tests/datasets/datasets-lua-02/dataset-dns.lua

```diff
@@ -0,0 +1,32 @@
+function init (args)
+    local needs = {}
+    needs["dns.request"] = tostring(true)
+    return needs
+end
+
+function thread_init (args)
+    dns_new, err = dataset.get_ref("dns-seen")
+    if err ~= nil then
+        SCLogWarning("dataset warning: " .. err)
+        return 0
+    end
+end
+
+function match (args)
+    ipver, srcip, dstip, proto, sp, dp = SCFlowTuple()
+    query = DnsGetDnsRrname()
+    if query == nil then
+        return 0
+    end
+    str = ipver .. ":<" .. srcip .. ">:<" .. dstip .. ">:" .. dp .. "--" .. query
+
+    ret, err = dataset.add(dns_new, str, #str);
```

Comment:
I think idiomatic Lua is more likely see us a method on `dns_new`, for example, looking at the `io` library you'll seem something like this for writing to files:
```
local f = io.open("filename", "w")
f:write("hello world")
```
Not really saying one is better than the other, just commenting on what I see in the standard library.